includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -554,6 +570,11 @@ public function process_payment( $order_id ) {
 				} catch ( Amount_Too_Small_Exception $e ) {
 					// This code would only be reached if the cache has already expired.
 					throw new Exception( WC_Payments_Utils::get_filtered_error_message( $e ) );
+				} catch ( Exception $e ) {
+					// If something goes on with API, forget the cached intent as we have no clues what exactly is wrong.
+					// It might be connectivity issues, intent state not allowing the update or post update period.
+					self::remove_upe_payment_intent_from_session();
```

Comment:
Keeping this for now. If the order changes status when an exception happens here, we can drop this statement.

Reply:
This is a must-handling. Reiterated with 82dfe6b68c304fc92f3e755bb67cd64cf07f0d86